### ISSUE TITLE
ArgumentListWrappingRule is confused inside some lambda blocks

### ISSUE THREAD

**Issue by ephemient** [pinterest/ktlint#1112]
## Expected Behavior

```kotlin
fun main() {
    Unit.run {
        println(
            "hello, world!"
        )
    }
}
```
should not report any indentation errors.

## Observed Behavior

```
test.kt:4:13: Unexpected indentation (expected 8, actual 12) (experimental:argument-list-wrapping)
test.kt:5:9: Unexpected indentation (expected 4, actual 8) (experimental:argument-list-wrapping)
```

## Your Environment

* Version of ktlint used: 0.41.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): command line
* Version of Gradle used (if applicable):
* Operating System and version: Linux 5.11
* Link to your project (if it's a public repository):
